sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -305,6 +305,9 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
               throw QueryCompilationErrors.invalidStarUsageError(operator.nodeName, Seq(s))
             }
 
+          case UnresolvedWindowExpression(_, WindowSpecReference(windowName)) =>
```

Comment:
@cloud-fan I am reluctant to remove the existing `UnresolvedWindowExpression` checks to avoid correctness issues.